src/vsr/replica.zig

```diff
@@ -7955,6 +7957,16 @@ pub fn ReplicaType(
             assert(!self.journal.has_prepare(message.header));
             assert(message.header.op > self.commit_min);
 
+            if (self.release.value < message.header.release.value and
+                self.replica == message.header.replica)
+            {
+                // Don't replica messages on a newer release than us if we were the one who
+                // originally sent it. This can happen if our release backtracked due to being
+                // reformatted.
+                log.warn("{}: replicate: ignoring prepare from newer release", .{self.replica});
+                return;
+            }
```

Comment:
This seems like a hack, but I didn't see a solution that is _less_ of a hack! The problem is that reformatting might regress our version, then we replicate a prepare which we previously originated (`message.header.replica == self.replica`). This trips:

```
            // Compare the release in this message to ours only if we authored the message.
            if (message.header.replica == self.replica) {
                assert(message.header.release.value <= self.release.value);
            }
```

in `send_message_to_replica_base`.